### PR TITLE
MQE-1633 + MQE-1514

### PR DESCRIPTION
**Pull request by KevinBKozan**
## Description
- Added AllureHelper class + unit tests
- Added calls to above class in MagentoWebDriver `see/dontsee` url actions + `CurlHandler` for persistence actions

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/verification tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)
 - [ ] Changes to Framework doesn't have backward incompatible changes for tests or have related Pull Request with fixes to tests

### CODE REVIEW

File: src/Magento/FunctionalTestingFramework/Allure/AllureHelper.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        Allure::lifecycle()->fire(new AddAttachmentEvent($data, $caption));
+    }
+
+    /**
+     * Adds Attachment to the last executed step. Required due to Allure root-step behavior
+     * @param mixed  $data
+     * @param string $caption
+     * @return void
+     */
+    public static function addAttachmentToLastStep($data, $caption)
```
Comment (tomreece) [inline src/Magento/FunctionalTestingFramework/Allure/AllureHelper.php]: Just curious about in general when to use this over the current step? I see you use this one in the curl handler, but I cant figure out why. Maybe add a bit to the phpdoc like:

`This function should be used when ___blahblahblah___`

**Comment by tomreece** [comment]
I do not see the request/response data in the build: http://10.234.204.237:8080/job/Functional-Tests-CE/2021/allure/#suites/8187718e2fd144987673ef34ee62b41b/136f33b4de6d9efe/

Am I blind?

**Comment by KevinBKozan** [comment]
@tomreece I forgot to commit a file for unit tests, that will fix the errors.

And let me retrigger/reset up branches for the build, I may have messed something up there.

File: src/Magento/FunctionalTestingFramework/Allure/AllureHelper.php (hunk 1 above)
Comment (KevinBKozan) [reply src/Magento/FunctionalTestingFramework/Allure/AllureHelper.php]: I'll add a comment to clear it up, the gist of it is that during a step (aka a function called in the test via `$I->doathing()`) you use current. If the framework is doign its own logic outisde a function like that, you can add artifact to the last step (which is usually a comment step)

**Comment by tomreece** [comment]
Looks good to me.

**Pull request merged**